go/libraries/doltcore/schema/typeinfo/typeinfo_test.go

```diff
@@ -0,0 +1,381 @@
+							}
+						})
+					}
+				})
+				require.True(t, atLeastOneValid, `all values reported false for "%v"`, ti.String())
+			}
+		})
+	}
+}
+
+// each TypeInfo in tiArrays is unique, so all equality comparisons should fail when the indices don't match
+func testTypeInfoEquals(t *testing.T, tiArrays [][]TypeInfo) {
+	for tiArrayIndex, tiArray := range tiArrays {
+		t.Run(tiArray[0].GetTypeIdentifier().String(), func(t *testing.T) {
+			// check this TypeInfo against its own variations, EX: Int16 & Int32
+			// a != b should also mean b != a
+			for i := range tiArray {
+				ti1 := tiArray[i]
+				t.Run(ti1.String(), func(t *testing.T) {
+					for j := range tiArray {
+						ti2 := tiArray[j]
+						t.Run(fmt.Sprintf(ti2.String()), func(t *testing.T) {
+							equality := ti1.Equals(ti2)
+							if i == j {
+								assert.True(t, equality)
+							} else {
+								assert.False(t, equality)
+							}
+						})
+					}
+				})
+			}
+			// we just check the first element and assume it'll hold true for the other values
+			firstTi := tiArray[0]
+			t.Run(fmt.Sprintf(`%v Others`, firstTi), func(t *testing.T) {
+				// check this TypeInfo against the other types, EX: Int16 & Float64
+				for tiArrayIndex2, tiArray2 := range tiArrays {
+					if tiArrayIndex == tiArrayIndex2 {
+						// this is the for loop above
+						continue
+					}
+					for _, otherTi := range tiArray2 {
+						t.Run(fmt.Sprintf(otherTi.String()), func(t *testing.T) {
+							equality := firstTi.Equals(otherTi)
+							assert.False(t, equality)
+						})
+					}
+				}
+			})
+		})
+	}
+}
+
+// ConvertNomsValueToValue and FormatValue should fail if the kind does not match the TypeInfo kind
+func testTypeInfoForeignKindHandling(t *testing.T, tiArrays [][]TypeInfo, vaArrays [][]types.Value) {
+	for _, tiArray := range tiArrays {
+		t.Run(tiArray[0].GetTypeIdentifier().String(), func(t *testing.T) {
+			for _, ti := range tiArray {
+				t.Run(ti.String(), func(t *testing.T) {
+					for _, vaArray := range vaArrays {
+						for _, val := range vaArray {
+							t.Run(fmt.Sprintf(`types.%v(%v)`, val.Kind().String(), val.HumanReadableString()), func(t *testing.T) {
+								if ti.NomsKind() != val.Kind() {
+									_, err := ti.ConvertNomsValueToValue(val)
+									assert.Error(t, err)
+									_, err = ti.FormatValue(val)
+									assert.Error(t, err)
+								}
+							})
+						}
+					}
+				})
+			}
+		})
+	}
+}
+
+// assuming valid data, verifies that the To-From string functions can round trip
+func testTypeInfoFormatParseRoundTrip(t *testing.T, tiArrays [][]TypeInfo, vaArrays [][]types.Value) {
+	for rowIndex, tiArray := range tiArrays {
+		t.Run(tiArray[0].GetTypeIdentifier().String(), func(t *testing.T) {
+			//TODO: determine the storage format for DecimalType
+			//TODO: determine the storage format for VarBinaryType
+			if tiArray[0].GetTypeIdentifier() == DecimalTypeIdentifier || tiArray[0].GetTypeIdentifier() == VarBinaryTypeIdentifier {
```

Comment:
Seems like you should probably just take these out of the types array until we can actually support them.